manager/allocator/networkallocator/networkallocator.go

```diff
@@ -303,38 +324,84 @@ func OnInit(options *ServiceAllocationOpts) {
 // IsServiceAllocated returns if the passed service has its network resources allocated or not.
 // init bool indicates if the func is called during allocator initialization stage.
 func (na *NetworkAllocator) IsServiceAllocated(s *api.Service, flags ...func(*ServiceAllocationOpts)) bool {
-	var options ServiceAllocationOpts
+	return !na.IsServiceToBeAllocated(s, flags...)
+}
```

Comment:
Feel free to change it. Personally, I don't really like the name `IsServiceAllocated`, since the function is really checking if the service *needs* allocation, not if it is allocated. I don't mind whether the function returns whether the service needs allocation or does not need allocation, as long as the name is clear.